x/merkledb/proof.go

```diff
@@ -644,7 +642,7 @@ func (proof *ChangeProof) Empty() bool {
 // If [start] is nil, there is no lower bound on acceptable keys.
 // If [end] is nil, there is no upper bound on acceptable keys.
 // If [kvs] is empty, returns nil.
-func verifyKeyChanges(kvs []KeyChange, start, end []byte) error {
+func verifyKeyChanges(kvs []KeyChange, start []byte, end Maybe[[]byte]) error {
```

Comment:
could we update the comment for this fn to refer to Nothing instead of nil as appropriate?